core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -35,6 +35,7 @@
 #include <impl/Kokkos_CheckedIntegerOps.hpp>
 #include <impl/Kokkos_DeviceManagement.hpp>
 #include <impl/Kokkos_ExecSpaceManager.hpp>
+#include <impl/Kokkos_Space_Instance.hpp>
```

Comment:
@dalg24 I don't know where to put this new macro, so anywhere you like is fine with me :smile:

Reply:
I am not convinced having a macro is a good idea here nor that we should dictate that individual backends have/use `verify_is_initialized`.

Reply:
* Instead of a macro, would you prefer a CRTP approach? Kokkos could have several CRTP classes that internal space classes could inherit from and so enforce some sorts of behaviors. But that approach seems quite difficult to implement...
* I guess the current PR does not enforce all internal space classes to have `verify_is_initialized`, it just proposes that those currently having it use the same implementation :smile:

Side question: `verify_is_initialized` returns a `bool`, but in fact the return is `true` or it aborts, so the return value is kind of useless...

Reply:
I would add a death test that potentially checks the error message with a regex.

Then I would delegate to backends developers and tell them to do whatever they want but that the test needs to pass.  I am not convinced that there is a need for a shared facility to achieve the desired behavior.  If I was maintaining a backend I would have gotten rid of that `verify_is_initialized` function a long time ago.  I don't like anything about it.